ibm/service/kubernetes/resource_ibm_container_cluster.go

```diff
@@ -980,19 +980,24 @@ func resourceIBMContainerClusterUpdate(d *schema.ResourceData, meta interface{})
 		// with major and minor updates.
 		updateAllWorkers := d.Get("update_all_workers").(bool)
 		if updateAllWorkers || d.HasChange("patch_version") || d.HasChange("retry_patch_version") {
-			workerFields, err := wrkAPI.List(clusterID, targetEnv)
+			workerFields, err := csClientV2.Workers().ListClassicWorkers(clusterID, false, targetEnvV2)
 			if err != nil {
 				return fmt.Errorf("[ERROR] Error retrieving workers for cluster: %s", err)
 			}
 
 			waitForWorkerUpdate := d.Get("wait_for_worker_update").(bool)
 
 			for _, w := range workerFields {
+				workerPool, err := csClient.WorkerPools().GetWorkerPool(clusterID, w.PoolID, targetEnv)
+				if err != nil {
+					return fmt.Errorf("[ERROR] Error retrieving worker pool: %s", err)
+				}
+
 				/*kubeversion update done if
 				1. There is a change in Major.Minor version
 				2. Therese is a change in patch_version & Traget kube patch version and patch_version are same
 				*/
-				if w.KubeVersion != w.TargetVersion {
+				if w.KubeVersion.Actual != w.KubeVersion.Target || w.LifeCycle.ActualOperatingSystem != workerPool.OperatingSystem {
```

Comment:
what is this change for
```
w.LifeCycle.ActualOperatingSystem != workerPool.OperatingSystem 
```

Reply:
So there are 2 cases when you need to call the worker update/replace API, one is if there is a newer BOM version, so the `w.KubeVersion.Actual != w.KubeVersion.Target` and the other is when you set the operating system of the workerpool to something else than what is currently the operating system. This is the code that will do the actual change to the worker's operating system, and here we do wait for the update to complete